lib/mongoid/contextual/mongo/pluck_enumerator.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Contextual
+    class Mongo
+
+      # Utility class to add enumerable behavior for
+      # Criteria#pluck_each. Should not be directly instantiated
+      # outside of Criteria#pluck_each.
+      #
+      # @api private
+      class PluckEnumerator
+        include Enumerable
+
+        # Create the new PluckEnumerator.
+        #
+        # @api private
+        #
+        # @example Initialize a PluckEnumerator.
+        #   PluckEnumerator.new(klass, view, fields)
+        #
+        # @param [ Class ] klass The base of the binding.
+        # @param [ Mongo::Collection::View ] view The Mongo view context.
+        # @param [ String, Symbol ] *fields Field(s) to pluck,
+        #   which may include nested fields using dot-notation.
+        def initialize(klass, view, fields)
+          @klass = klass
+          @view = view
+          @fields = fields
+        end
+
+        # Iterate through plucked field value(s) from the database
+        # for the view context. Yields result values progressively as
+        # they are read from the database. The yielded results are
+        # normalized according to their Mongoid field types.
+        #
+        # @api private
+        #
+        # @example Iterate through the plucked values from the database.
+        #   context.pluck_each(:name) { |name| puts name }
+        #
+        # @param [ Proc ] block The block to call once for each plucked
+        #   result.
+        #
+        # @return [ Enumerator, PluckEnumerator ] The enumerator, or
+        #   self if a block was given.
+        def each(&block)
+          return to_enum unless block_given?
+
+          @view.projection(normalized_field_names.index_with(true)).each do |doc|
+            yield_result(doc, &block)
+          end
+
+          self
+        end
+
+        private
+
+        def database_field_names
+          @database_field_names ||= @fields.map {|f| @klass.database_field_name(f) }
+        end
+
+        def normalized_field_names
+          @normalized_field_names ||= if Mongoid.legacy_pluck_distinct
+                                        database_field_names
+                                      else
+                                        @fields.map {|f| @klass.cleanse_localized_field_names(f) }
+                                      end
```

Comment:
Your implementation of `normalized_field_names` differs slightly to the one I had previously. 

Can you add a test in which we pluck both a localized field, and it's _transaltions hash?

For example: 
```
Dictionary.pluck(:description, :description_translations)
```
and verify that this new implementation correctly returns 
```
[["t", {"en" => "t", "de" => "u" }]]
```
for current locale being `:en`

Reply:
@Neilshweky this test is [already in this PR](https://github.com/mongodb/mongoid/pull/5332/files/91df1608b47256be1931a0b86afdb284e7f367f5#diff-56ee0ff139cc683178654f38c4899e8b2c716c60c79c8c0a681ed76fa997526aR2619) for both `#pluck` and `#pluck_each`, legacy and non-legacy cases (I copied all the tests you wrote for #pluck.)

Github [collapses the diff](https://github.com/mongodb/mongoid/pull/5332/files/91df1608b47256be1931a0b86afdb284e7f367f5#diff-56ee0ff139cc683178654f38c4899e8b2c716c60c79c8c0a681ed76fa997526a) for `criteria_spec.rb` b/c it is large so you may have missed it.

Reply:
Okay great!